examples/applications/rag-aws/chatbot.yaml

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+topics:
+  - name: "questions-topic"
+    creation-mode: create-if-not-exists
+  - name: "answers-topic"
+    creation-mode: create-if-not-exists
+errors:
+    on-failure: "skip"
+pipeline:
+  - name: "convert-to-structure"
+    type: "document-to-json"
+    input: "questions-topic"
+    configuration:
+      text-field: "question"
+  - name: "compute-embeddings"
+    type: "compute-ai-embeddings"
+    configuration:
+      model: "amazon.titan-embed-text-v1"
+      embeddings-field: "value.embeddings_vector"
+      text: "{{ value.question }}"
+      batch-size: 10
+      flush-interval: 500
+  - name: "lookup-related-documents"
+    type: "query-vector-db"
+    configuration:
+      datasource: "OpenSearch"
+      query: |
+        {
+          "index": "rag-index-example-3",
+          "size": 10,
+          "query": {
+            "knn": {
+              "embeddings": {
+                "vector": ?,
+                "k": 10
+              }
+            }
+          }
+        }
+      fields:
+        - "value.embeddings_vector"
+      output-field: "value.related_documents"
+  - name: "ai-chat-completions"
```

Comment:
Is it possible to add re-rank here? 
In order to do it we have to also get the embeddings for each retrieved document.

re-rank is very important because it allows you to select more docs from the database and then refine the search

Reply:
done, thanks